pkg/controller/vsphere/reconciler_test.go

```diff
@@ -1344,6 +1345,44 @@ func TestReconcileMachineWithCloudState(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	createAndAttachTag := func(tagName, categoryName string) {
+		if err := session.WithRestClient(context.TODO(), func(c *rest.Client) error {
+			tagsMgr := tags.NewManager(c)
+
+			categoryID, err := tagsMgr.CreateCategory(context.TODO(), &tags.Category{
+				AssociableTypes: []string{"VirtualMachine"},
+				Cardinality:     "SINGLE",
+				Name:            categoryName,
+			})
+			if err != nil {
+				return err
+			}
+
+			tagID, err := tagsMgr.CreateTag(context.TODO(), &tags.Tag{
+				CategoryID: categoryID,
+				Name:       tagName,
+			})
+			if err != nil {
+				return err
+			}
+
+			err = tagsMgr.AttachTag(context.TODO(), tagID, vmObj.Reference())
+			if err != nil {
+				return err
+			}
+
+			return nil
+		}); err != nil {
+			t.Fatal(err)
+		}
+	}
```

Comment:
I feel like I've seen this before in another PR, could this be a proper function rather than an anonymous one that multiple tests could use?